contracts/src/bridge/Bridge.sol

```diff
@@ -55,14 +62,71 @@ contract Bridge is OwnableUpgradeable, DelegateCallAware, IBridge {
         return outbox;
     }
 
-    function allowedInboxes(address inbox) external view override returns (bool) {
-        return allowedInboxesMap[inbox].allowed;
+    function allowedDelayedInboxes(address inbox) external view override returns (bool) {
+        return allowedDelayedInboxesMap[inbox].allowed;
     }
 
     function allowedOutboxes(address outbox) external view override returns (bool) {
         return allowedOutboxesMap[outbox].allowed;
     }
 
+    /// @param batchPoster sequencer that included the batch, or zero if not a sequencer
+    function enqueueSequencerMessage(
+        bytes32 dataHash,
+        uint256 afterDelayedMessagesRead,
+        address batchPoster
+    )
+        external
+        override
+        returns (
+            uint256 seqMessageCount,
+            bytes32 beforeAcc,
+            bytes32 delayedAcc,
+            bytes32 acc
+        )
+    {
+        if (msg.sender != sequencerInbox) revert NotSequencerInbox(msg.sender);
+        seqMessageCount = sequencerInboxAccs.length;
+        if (sequencerInboxAccs.length > 0) {
+            beforeAcc = sequencerInboxAccs[sequencerInboxAccs.length - 1];
+        }
+        if (afterDelayedMessagesRead > 0) {
+            delayedAcc = delayedInboxAccs[afterDelayedMessagesRead - 1];
+        }
+        acc = keccak256(abi.encodePacked(beforeAcc, dataHash, delayedAcc));
+        sequencerInboxAccs.push(acc);
+
+        if (batchPoster != address(0)) {
+            // we ignore the return value as we don't need the updated delayed inbox count
+            // as this msg isn't included in the current sequencer batch
+            submitBatchSpendingReport(batchPoster, dataHash, seqMessageCount);
+        }
+    }
+
+    function submitBatchSpendingReport(
+        address batchPoster,
+        bytes32 dataHash,
+        uint256 seqMessageCount
+    ) internal returns (uint256) {
+        bytes memory messageData = abi.encodePacked(
+            block.timestamp,
+            batchPoster,
+            dataHash,
+            seqMessageCount,
+            block.basefee
+        );
+
+        return
+            addMessageToDelayedAccumulator(
```

Comment:
We'll need to emit the message data in an inbox-like event somehow. Maybe it'd be better for the sequencer inbox to still submit the batch spending report?

Reply:
pushed a change for the sequencer inbox to do so